mldsa/native/x86_64/src/poly_use_hint_88_avx2.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) The mldsa-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+/*
+ * This file is derived from the public domain
+ * AVX2 Dilithium implementation @[REF_AVX2].
+ */
+
+#include "../../../common.h"
+
+#if defined(MLD_ARITH_BACKEND_X86_64_DEFAULT) && \
+    !defined(MLD_CONFIG_MULTILEVEL_NO_SHARED)
+
+#include <immintrin.h>
+#include <stdint.h>
+#include <string.h>
+#include "arith_native_x86_64.h"
+#include "consts.h"
+
+#define _mm256_blendv_epi32(a, b, mask)                        \
+  _mm256_castps_si256(_mm256_blendv_ps(_mm256_castsi256_ps(a), \
+                                       _mm256_castsi256_ps(b), \
+                                       _mm256_castsi256_ps(mask)))
+
+void mld_poly_use_hint_88_avx2(__m256i *b, const __m256i *a,
+                               const __m256i *hint)
+{
+  unsigned int i;
+  __m256i f, f0, f1, h, t;
+  const __m256i q =
+      _mm256_load_si256(&mld_qdata.vec[MLD_AVX2_BACKEND_DATA_OFFSET_8XQ / 8]);
+  const __m256i hq = _mm256_srli_epi32(q, 1);
+  /* check-magic: 11275 == round((2**24*128) / ((MLDSA_Q - 1) / 44)) */
+  const __m256i v = _mm256_set1_epi32(11275);
+  const __m256i alpha = _mm256_set1_epi32(2 * MLDSA_GAMMA2);
+  const __m256i off = _mm256_set1_epi32(127);
+  const __m256i shift = _mm256_set1_epi32(128);
+  const __m256i max = _mm256_set1_epi32(43);
+  const __m256i zero = _mm256_setzero_si256();
+
+  for (i = 0; i < MLDSA_N / 8; i++)
+  {
+    f = _mm256_load_si256(&a[i]);
+    h = _mm256_load_si256(&hint[i]);
+
+    /* Reference: The reference avx2 implementation calls poly_decompose to
+     * compute all a1, a0 before the loop. Here we compute them within the loop
+     * to reduce stack usage.
```

Comment:
```suggestion
    /* Reference: The reference avx2 implementation calls poly_decompose to
     * compute all a1, a0 before the loop. 
```

You can remove the second sentence. The compiler may optimize that stack usage away anyway. The reason we cannot call decompose here is that decompose is in a separate file.

Reply:
Fixed. Thanks for the suggestion!